sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java

```diff
@@ -497,7 +497,7 @@ private void ensureWindowsAreASet() {
       // right, and cast the window type away here.
       @SuppressWarnings({"unchecked", "rawtypes"})
       Coder<Collection<? extends BoundedWindow>> collectionCoder =
-          (Coder) CollectionCoder.of(this.windowCoder);
+          (Coder) ListCoder.of(this.windowCoder);
```

Comment:
is there any particular reason why CollectionCoder was changed to ListCoder ?

Reply:
Collections aren't required to be ordered (e.g. HashSet). It would be valid to return any collection which maintains duplicates such as a multiset that doesn't maintain order. Swapping to a ListCoder ensures that we maintain the ordering of windows.